src/hardware/driver/internal_adc.rs

```diff
@@ -76,8 +85,8 @@ impl InternalAdc {
                 self.output_current.1.read_normalized().unwrap()
             }
         };
-        const SCALE: f32 = V_REF / R_SENSE; // Current sense scale       ToDo
-        const OFFSET: f32 = 0.0; // Current sense offset         ToDo
-        (ratio + OFFSET) * SCALE
+        let scale =
+            V_REF * I_SCALE / self.variant[ch as usize].transimpedance(); // current sense scale
```

Comment:
Invert the function return to avoid the division.
And push `V_REF*I_SCALE` into the (single-use) method.

Reply:
I would leave those constants here since they have to do with the ADC and current sensing. At least I will surely forget that the ADC reference voltage is to be found in the `ChannelVariant`.

Reply:
Ok. But do invert the method. `current_sense_{transconductance,scale,gain}()` maybe.

Reply:
its `current_sense_admittance` now